packages/react/src/data-table/DataTable.tsx

```diff
@@ -0,0 +1,148 @@
+      pagination: {
+        pageIndex,
+        pageSize,
+      },
+      sorting,
+    },
+  });
+
+  let offset = 0;
+  const offsets = Object.fromEntries(
+    table
+      .getAllColumns()
+      .flatMap((column) => [
+        [column.id, (offset += column.getSize()) - column.getSize()],
+      ]),
+  );
+
+  return (
+    <Box alignItems="center" display="flex" flexDirection="column">
+      <Table>
+        <TableHead {...styles.tableHeader()}>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <TableRow key={headerGroup.id}>
+              {headerGroup.headers.map((header) => {
+                return (
+                  <TableHeaderCell
+                    key={header.id}
+                    style={{
+                      ...assignInlineVars({
+                        [styles.cellOffset]: header.column.getIsPinned()
+                          ? `${offsets[header.column.id]}px`
+                          : undefined,
+                        [styles.columnWidth]: `${header.getSize()}px`,
+                      }),
+                    }}
+                    {...styles.tableHead({
+                      pinned: header.column.getIsPinned() ?? undefined,
+                    })}
+                  >
+                    {header.column.columnDef.header &&
+                    typeof header.column.columnDef.header !== "string" ? (
+                      createElement(
+                        header.column.columnDef.header,
+                        header.getContext(),
+                      )
+                    ) : (
+                      <DataTableHeader {...header.getContext()}>
+                        {header.column.columnDef.header}
+                      </DataTableHeader>
+                    )}
+                  </TableHeaderCell>
+                );
+              })}
+            </TableRow>
+          ))}
+        </TableHead>
+        <TableBody>
+          {table.getRowModel().rows.map((row) => (
+            <TableRow key={row.id}>
+              {row.getVisibleCells().map((cell) => {
+                return (
+                  <TableCell
+                    key={cell.id}
+                    style={{
+                      ...assignInlineVars({
+                        [styles.cellOffset]: cell.column.getIsPinned()
+                          ? `${offsets[cell.column.id]}px`
+                          : undefined,
+                        [styles.columnWidth]: `${cell.column.getSize()}px`,
+                      }),
+                    }}
+                    {...styles.tableCell({
+                      pinned: cell.column.getIsPinned() ?? undefined,
+                    })}
+                  >
+                    {flexRender(cell.column.columnDef.cell, cell.getContext())}
+                  </TableCell>
+                );
+              })}
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+
+      {data.length > pageSize && (
```

Comment:
let's re-use table state
```suggestion
      {data.length > table.getState().pagination.pageSize && (
```

Reply:
the reason being the pagination state can be controlled or uncontrolled - and we need to write code that is agnostic of that